### PR TITLE
Add build and publish config for CommonJS

### LINKED ISSUE #6: Publish commonJS modules
I prefer ES modules, but unfortunately I can't use them in my current project, because react only has commonJS exports. Can you publish the package containing both ESM and CJS modules? That way I can use it in my project.

### PR DESCRIPTION
**Pull request by desko27**
Fixes #6

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,15 +1,27 @@
 {
   "name": "react-call",
-  "version": "1.1.0",
+  "version": "1.2.0-0",
   "description": "Call your React components",
+  "license": "MIT",
+  "repository": "desko27/react-call",
   "author": "Ismael Ramon <desko27@gmail.com> (https://desko.dev)",
   "type": "module",
```
Comment (mariusGundersen) [inline package.json:8]: This line needs to be removed for it to work.

File: package.json [hunk 2]
```diff
@@ -1,15 +1,27 @@
...
-  "main": "dist/main.js",
+  "exports": {
+    ".": {
+      "default": "./dist/main.js",
```
Comment (mariusGundersen) [inline package.json:11]: I think this might need to be changed to `"import": "./dist/main.js"`

File: package.json [hunk 3]
```diff
@@ -1,15 +1,27 @@
...
+      "require": "./dist/main.cjs",
+      "types": "./dist/main.d.ts"
+    }
+  },
+  "main": "./dist/main.js",
```
Comment (mariusGundersen) [inline package.json:16]: You might also need to change this to 
```
"main": "./dist/main.cjs",
"module": "./dist/main.js"
```

**Review by mariusGundersen — CHANGES_REQUESTED** [review]
According to [this article](https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html) and others I have read you need to remove the `"type": "module"` line for it to work. I tried with those changes locally at that made it work for me.

**Comment by desko27** [comment]
Hi @mariusGundersen, thanks a lot for your involvement!

I see, having the `main.cjs` file as main is the "CJS first" approach. I was trying an "ESM first" approach, but I can see that's not gonna work. I didn't realize [even Vite.js recommends CJS first](https://vitejs.dev/guide/build#library-mode).

But they also recommend keeping `"type": "module"` and I'm actually getting build errors if I remove it. I've published `react-call@next` again with your suggested approach but keeping type module. Could you check if that works for you? Thanks for your patience.

**Comment by mariusGundersen** [comment]
Yes, now it works! It seems like the problem was `"default": "..."`, it works much better with `"import": "..."`, probably because the default is controlled by `"type": "module"`.

**Comment by mariusGundersen** [comment]
Thank you for taking the time to fix this for me 🥳

**Comment by desko27** [comment]
Glad to hear that! I've also been reading a bit more about it just to make sure. According to [Vite.js CJS troubleshooting](https://vitejs.dev/guide/troubleshooting#cjs) what we're doing matches the "Configure ESM as default, opt-in to CJS if needed" approach, which I think is great to remain in the modern direction!

I'll check everything is still working properly for ESM consumers and we'll get this published on v1.2.0.

@mariusGundersen thanks for contributing!

**Pull request merged**